src/Uno.UI.Runtime.Skia.Gtk/UI/Xaml/Controls/TextBoxViewExtension.cs

```diff
@@ -261,5 +273,15 @@ private void ContentElementSizeChanged(object sender, Windows.UI.Xaml.SizeChange
 			UpdateSize();
 			UpdatePosition();
 		}
+
+		public void SetIsPassword(bool isPassword)
+		{
+			if (_currentInputWidget is Entry entry)
+			{
+				entry.Visibility = !isPassword;
+				// InputPurpose doesn't have any visual effects.
+				entry.InputPurpose = InputPurpose.Password;
```

Comment:
Should this also account for `isPassword = true`?

Reply:
I didn't see any visual effects for it, and don't know what exactly does it change. So I preferred keeping `Password` since this is still the purpose.

Reply:
From https://valadoc.org/gtk+-3.0/Gtk.InputPurpose.html

> This information is useful for on-screen keyboards and similar input methods to decide which keys should be presented to the user.

I don't think this should change when clicking the reveal password button.

Reply:
Is it necessary to set `InputPurpose` here at all?

Reply:
The only behavior change is how on-screen keyboards decide to show, which I assume is good to have them know it's a password input.

Reply:
What I mean is, won't we have already set `InputPurpose = InputPurpose.Password` when the `Entry` widget was created?

Reply:
Oh indeed. Will remove it. Thanks!